### PR TITLE
Feature/cypress testing cag

### LINKED ISSUE #15: Cypress Testing - end to end testing

### PR DESCRIPTION
**Pull request by cagallo**
## What Features are Being Added?

- Created fixtures to test favorited books (cypress/fixtures)
- Cypress test favorites page(cypress/integration/favorites_page_spec.js)
- Cypress test book details page(cypress/integration/book_details_spec.js)
- Cypress test error page(cypress/integration/error_page_spec.js)

## What is Being Refactored?

- Edited previous tests to reflect the change in type case, change intercept to isbn (instead of id-book details page)
- Cleaned up spacing 

## Where should the reviewer start?

- cypress/integration

## How should this be manually tested?

-npm i, npm run cypress, click on each test

## What are the relevant tickets?

- test favorites page user flow, error page user flow, book details page user flow

## Any Other Notes/Questions/Problems?

- None, don't love our test for the Add to Favorites button but it is passing. We can try adding the delete from favorites although may be more of the same as far as having issues with the rendering because the state doesn't change in our tests.

### CODE REVIEW

**Review by ectweitmann — APPROVED** [review]
This looks great! The only thing I found is we need to add anonymous functions to the handle clicks in the determineButton function. We were getting a memory leak in cypress but that should resolve it.

File: cypress/integration/book_details_spec.js [hunk 1]
```diff
@@ -1,14 +1,45 @@
 describe('Book Club book details page user flow', () => {
 
   it('should display all book details for a single book', () => {
-    cy.intercept('https://book-club-api-2110.herokuapp.com/api/v1/books/2', { fixture: 'singleBook.json' })
-    cy.visit('http://localhost:3000/2')
-      .get('img').should('have.attr', 'src', 'https://storage.googleapis.com/du-prd/books/images/9780525559474.jpg')
-      .get('.selected-title').contains('THE MIDNIGHT LIBRARY')
+    cy.intercept('https://book-club-api-2110.herokuapp.com/api/v1/books/9780525559474', { fixture: 'singleBook.json' })
```
Comment (krdurrett) [inline cypress/integration/book_details_spec.js:4]: Nice work on this book detail test

File: cypress/integration/book_details_spec.js [hunk 2]
```diff
@@ -1,14 +1,45 @@
...
+       body: {
+          "isbn": "9780525559474",
+          "title": "The Midnight Library",
+          "description": "Nora Seed finds a library beyond the edge of the universe that contains books with multiple possibilities of the lives one could have lived.",
+          "amazon_link": "https://www.amazon.com/dp/0525559477?tag=NYTBSREV-20",
+          "book_image": "https://storage.googleapis.com/du-prd/books/images/9780525559474.jpg",
+          "author": "Matt Haig"
+        }
+     })
+    cy.intercept('PATCH', 'https://book-club-api-2110.herokuapp.com/api/v1/books/9780525559474', { "isFavorited": "true" })
```
Comment (krdurrett) [inline cypress/integration/book_details_spec.js:28]: So many intercepts! But glad they are working now

File: cypress/integration/error_page_spec.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+describe('Book Club error page user flow', () => {
```
Comment (Fordo29) [inline cypress/integration/error_page_spec.js:1]: Thanks for writing out this test

File: cypress/integration/error_page_spec.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+  it('should display 404 page when visiting a bad link', () => {
```
Comment (krdurrett) [inline cypress/integration/error_page_spec.js:3]: Great test for the error page

File: cypress/integration/favorites_page_spec.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+describe('Book Club favorites page user flow', () => {
+
+  it('should display favorited book cards', () => {
```
Comment (krdurrett) [inline cypress/integration/favorites_page_spec.js:3]: Very thorough testing!

**Review by krdurrett — APPROVED** [review]
Such a productive struggle, very good work debugging

**Pull request merged**